pkg/internal/storage/repo/transactions.go

```diff
@@ -602,6 +603,49 @@ func (txs *Transactions) conditionsBySpec(spec *pkgentity.TransactionReadSpecifi
 	if spec.DescriptionContains != nil {
 		conditions = append(conditions, cmpCondition(table.Description, spec.DescriptionContains))
 	}
+	if spec.Labels != nil {
+		conditions = append(conditions, txs.labelConditions(ctx, spec.Labels)...)
+	}
 
 	return conditions
 }
+
+func (txs *Transactions) labelConditions(ctx context.Context, labels *pkgentity.ComparableSet[string]) []gen.Condition {
+	var conds []gen.Condition
+	table := &txs.query.Transaction
+	txl := &txs.query.TransactionLabel
+
+	if labels.Empty {
+		sub := txl.WithContext(ctx).
+			Select(txl.TransactionID).
+			Where(txl.TransactionID.EqCol(table.ID))
+
+		conds = append(conds,
+			field.Not(field.CompareSubQuery(field.ExistsOp, nil, sub.UnderlyingDB())),
+		)
+	}
```

Comment:
I think, there is no point to add ContainAny & ContainAll parts if `Empty` is on. I would just return the conds here.

Reply:
```suggestion
		return []gen.Condition{
			field.Not(field.CompareSubQuery(field.ExistsOp, nil, sub.UnderlyingDB())),
		}
```

sure changed to this ...